### ISSUE TITLE
Issue w/uploading image to Facebook

### ISSUE THREAD

**Issue by bluesky75** [ideashower/ShareKit#262]
Not sure if I have had one too many drinks over the last couple of days but for whatever reason, my images are not uploading to Facebook.  The image is local to the app sandbox itself, not retrieved from outside web service.  However, I didn't have any issues posting text. 

I was able to post an image, using the same UIImage object reference (along with text), to Twitter just fine.

I've done some scouring over the net to see if anyone else has had the same issue but i'm beginning to think it may just be me.  I've followed all of the instructions to utilize ShareKit within my app.  I also made sure I pulled the latest https://github.com/dagerydoo/ShareKit version and followed the readMe file in there.  All is great with that version...but when uploading an image...it simply doesn't appear any where on the wall nor albums, even though ShareKit displays a checkmark that all is good.

For kicks and giggles, I tried some other sample demo apps, even the one that comes with the Facebook Connect, and they post text fine...but not images.  I always seem to get a server response but no image appears on the wall or any library.

Any help on this issue would be much appreciated.

**Comment by tobytaro** [comment]
One of my app's user also meet this problem.
Do someone know how to solve it?

**Comment by bobbypage** [comment]
Try using the new FB Connect authentication system v2.0. Get this from my fork/pull request: https://github.com/ideashower/ShareKit/pull/138

**Comment by bluesky75** [comment]
Thanks bobbypage.  I did try using your fork.  Same results...I was able to post text status but not able to upload images.  I double checked the photo albums as well and nothing.  I also used the example provided within the fork and plugged in my FB app ID.  Same thing, can post text no problem but cannot upload images...even though ShareKit shows me it did. 

And to be certain, I re-tested with Twitter and no issues there with posting text and images.  Very frustrating.

**Comment by bluesky75** [comment]
Since this doesn't seem to be a widespread issue, is there something I may have missed on the Facebook side of things to allow for photos to be uploaded?  I'm grasping at straws here so this may be a fruitless question.

**Comment by bluesky75** [comment]
Well, once again I have tried all of the various demo apps using Facebook Connect and am still running into the same issue.  I can post text on wall without any problems.  When posting images (via UIImage object), they do not appear on the wall nor within any photo album.

Can anyone help or at least make a suggestion on what I should try?

**Comment by bobbypage** [comment]
Contact the Facebook Connect team?

**Comment by bluesky75** [comment]
Thanks for the response bobbypage.  I did contact the Facebook connect team and am still awaiting a response.  I guess I'm just frustrated because no one else seems to have this issue.  Just not sure if it's something so simple that I've overlooked or maybe even the way how my app is registered on Facebook itself.

**Comment by bobbypage** [comment]
Yeah, it's a really weird issue, facebook photo uploads work fine for me (using my fork).

A couple thing that you should try:
1. Register a different API key from Facebook
2. Try using the old (deprecated) FBConnect SDK: https://github.com/shazron/fbconnect-iphone

Hope it works,
-David

**Comment by bluesky75** [comment]
I think I'm going to go ahead and close this issue.  I appreciate those who responded and those who at least looked at the comments.  As it stands, I'm currently able to post text and images to Twitter, Tumbler, and Evernote...but text only to Facebook.  I've tried to contact Facebook Connect and haven't gotten a reply back.  I've also posted to the Facebook Connect dev forum site as well.  I'm literally stumped on this one and it's extremely frustrating.  but, I do appreciate the suggestions made.  Thanks.

**Comment by tobytaro** [comment]
The issue did exist! Some of my user tell me that they can't upload the image. It seems fine when sharing the in the app. And the message show "Saved". But the image can't be found on facebook!
But in my side, everything is fine!
I wonder if I use the same api key for 2 of my apps. One is full version, one is free version.

**Comment by CaryGao** [comment]
I have the same problem,Pls help!

**Comment by mosh1** [comment]
I'm suddenly having the same issue with my app as well..it has been working for months prior to that. Anyone else?

**Comment by blastStu** [comment]
I have this issue too!  The app reports everything is OK, but simply nothing shows up on FB.  The FB app is not in sandbox either...

**Comment by bluesky75** [comment]
Unfortunately, I haven't been able to narrow down the issue myself.  The only thing I can think of is it's between FB Connect and FB itself.  I know they both get wonky but I've never experienced anything like this before.  For the time being, I just left out any FB capability within my app.  I'm still trying to hunt down the issue but I don't know enough about FB connect to further narrow it down if at all possible.

**Comment by blastStu** [comment]
Hmm, it'd be a real shame to have to leave Facebook sharing out (but I too am running out of options).  Facebook is probably the best way to generate a load of traffic to the app!  I've just tried to share a simple text string and this didn't appear either...

**Comment by blastStu** [comment]
Perhaps I'm failing on this bit.  In dagerydoo's instructions, step 1 it says:

(or merge if you already use them, remembering to properly delegate)

What is this proper delegating that I'm supposed to remember to do?  Say my app is called Appy, then in AppyAppDelegate.m I pasted:
- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation 
  {
  return [SHKFacebook handleOpenURL:url];
  }
- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url 
  {
  return [SHKFacebook handleOpenURL:url];
  }

Oh, and also imported:

#import "SHKFacebook.h"

**Comment by bobbypage** [comment]
blastStu,

My advice, don't deal with the crappy facbook URL auth. You can disable it and use the good old normal pop up window.

You can disable it under Facebook.m.

I've disabled it in my fork [here](https://github.com/ideashower/ShareKit/pull/138)

**Comment by blastStu** [comment]
hmm, well. I had my plist set incorrectly, so with that done I managed to post a text update.

Then I went back to trying images, but it gets more weird.  I shorted the title of the image simply down to 'text' and this posted to my Facebook profile. Great I thought.. I went through trying different version of my title to try to find why it wouldn't post with no success, then went back to trying just 'text' again and it wouldn't post! Perhaps FB rate limits the amount of posts I can make in a certain amount of time?

bluesky75, did you have any punctuation in your string?  I had a comma and an exclamation mark, I wonder if these made a difference?  The string is 58 chars long...

**Comment by guitarpastdusk** [comment]
Could it be a FB user profile issue? I have been facing this issue for my FB profile for more than a month but when I created a new FB user and posted to that, it went through fine. Still havent managed to figure out why I cant post to my wall using my app but to other's walls. Havent tried this with text, only images.

**Comment by alessani** [comment]
I find out why sharing UIImage on Facebook apparently works only the first time: After uploading a UIImage to Facebook I removed it from my stream, so the following images didn't appear on the stream because I remove the post (the album post), but they were on my photo albums. Maybe someone could have the same problem.

**Comment by pomello** [comment]
same problem for me: first image upload works fine, than i remove the post from the wall.
following image uploads are not shown in the wall, but all images appear in the photo album...

**Comment by ronak28286** [comment]
@alessani : can you please let me know how did you manage to remove it from stream. I am facing same problem.

**Comment by ronak28286** [comment]
Please share me the code for uploading images to facebook wall.

Appreciate your help.

Thanks,

**Comment by prabhjottechblue** [comment]
Can anybody help me. I am new to sharekit and have a lil knowledge of objective C. So need some way out how can I retrieve my javascript variable in .m file in xcode.  

I am getting a newsFeed JSON object that contains a url and this url I need to pass to facebook share method so that I can make a post on Facebook. I will be thankful if somebody can illustrate this with an example.

Thanks
Prabhjot
